File: yht0827/1월/1월 12일/완주하지 못한 선수.java
import java.util.Arrays;

class Solution {
	public String solution(String[] participant, String[] completion) {
		Arrays.sort(participant);
		Arrays.sort(completion);

		int len = participant.length - 1;
		String answer =  participant[len];

		for (int i = 0; i < len; i++) {
			if (!participant[i].equals(completion[i])) {
				answer = participant[i];
				break;
			}
		}

		return answer;
	}
}

File: Eunleelee/2024-01/1월 12일/완주하지_못한_선수.py
# 처음에는 배열로 접근을 했는데 효율성 5번에서 문제가 발생
# 배열보다 접근 속도가 빠른 딕셔너리로 형 변환을 해줘서 해결

def solution(participant, completion):
    
    if len(participant) == 1:
        return participant[0]
    
    if len(set(participant)) - len(set(completion)) == 1:
        answer = list(set(participant) - set(completion))
        return answer[0]
    
    participant.sort()
    participant_di = {i : participant[i] for i in range(len(participant))}
    
    completion.sort()
    completion_di = {i : completion[i] for i in range(len(completion))}
    
    for key, value in completion_di.items():
        if value != participant_di[key]:
            return participant_di[key]
        
    return participant[-1]

print(solution(["mislav", "stanko", "mislav", "ana"],	["stanko", "ana", "mislav"]))